src/Microsoft.ComponentDetection.Detectors/pip/PipReportComponentDetector.cs

```diff
@@ -308,6 +339,54 @@ protected override async Task OnFileFoundAsync(ProcessRequest processRequest, ID
         }
     }
 
-    private bool IsPipReportManuallyDisabled()
-        => this.envVarService.IsEnvironmentVariableValueTrue(DisablePipReportScanEnvVar);
+    private async Task RegisterExplicitComponentsInFileAsync(
+        ISingleFileComponentRecorder recorder,
+        string filePath,
+        string pythonPath = null)
+    {
+        var initialPackages = await this.pythonCommandService.ParseFileAsync(filePath, pythonPath);
+        if (initialPackages == null)
+        {
+            return;
+        }
+
+        var listedPackage = initialPackages.Where(tuple => tuple.PackageString != null)
+            .Select(tuple => tuple.PackageString)
+            .Where(x => !string.IsNullOrWhiteSpace(x))
+            .Select(x => new PipDependencySpecification(x))
+            .Where(x => !x.PackageIsUnsafe())
+            .Where(x => x.PackageConditionsMet(this.pythonResolver.GetPythonEnvironmentVariables()))
+            .ToList();
+
+        listedPackage.Select(x => (x.Name, Version: x.GetHighestExplicitPackageVersion()))
+            .Where(x => !string.IsNullOrEmpty(x.Version))
+            .Select(x => new PipComponent(x.Name, x.Version))
+            .Select(x => new DetectedComponent(x))
+            .ToList()
+            .ForEach(pipComponent => recorder.RegisterUsage(pipComponent, isExplicitReferencedDependency: true));
```

Comment:
Note: this is just registering the usage of these components, not running the root resolution since that logic uses metadata from pypi. 

Not sure if there is anything else we can glean from the files themselves to record in the `ISingleFileComponentRecorder`?